ns1/data_source_record_test.go

```diff
@@ -0,0 +1,52 @@
+package ns1
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+
+	"gopkg.in/ns1/ns1-go.v2/rest/model/dns"
+)
+
+func TestAccDataSourceRecord(t *testing.T) {
+	var record dns.Record
+	dataSourceName := "data.ns1_record.test"
+	rString := acctest.RandStringFromCharSet(15, acctest.CharSetAlphaNum)
+	zoneName := fmt.Sprintf("terraform-test-%s.io", rString)
+	domainName := fmt.Sprintf("test.%s", zoneName)
+	rrType := "CNAME"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceResource(zoneName, domainName, rrType),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRecordExists("ns1_record.it", &record),
+					resource.TestCheckResourceAttr(dataSourceName, "zone", zoneName),
+					resource.TestCheckResourceAttr(dataSourceName, "domain", domainName),
+					resource.TestCheckResourceAttr(dataSourceName, "type", rrType),
```

Comment:
It looks like this test is currently only configuring attributes on the record resource that are also configurable on the data source.  I think we also want to verify that values for `computed` attributes in the data source (ie `answers`) are being populated correctly.

We should configure some additional properties on the record resource for this test and verify their values in the data source are correct.